### ISSUE TITLE
Refactor Report End to End tests to use a common base

### ISSUE THREAD

**Issue by jniles** [third-culture-software/bhima#3717]
While working on https://github.com/IMA-WorldHealth/bhima/pull/3716, it has become apparent that most of our page objects for reports are simply:

```js
 async closeCashflowReportPreview() {
   await this.page.closePreview();
 }
```

This doesn't make a lot of sense to me.  We are basically just renaming a function (`closePreview`) as `closeCashflowReportPreview`.  This happens in most tests, not just the cashflow report tests.

Virtually the only thing that changes is the configuration logic for the report.  This logic is set by what parameters the report takes.  For example:
 1. Start/End date
 2. Fiscal year selection
 3. Period selection
 4. Checkbox/Radio options
etc..

So really, all we need to do is write a new `configure()` method, and then we can inherit all the rest of the common functionality.

**Proposal**
I propose we create `BaseReportPage` that will be inherited from all other report tests with the `extend` keyword.  Like this:
```js
class CashflowReportPage extends BaseReportPage {
  constructor() {
    super();
  }
}
```
Then, instead of manually calling `this.page.closePreview()`, we can just call `CashflowReportPage.closePreview()`.  This is much nicer.

What about `printPreview()` and `saveReport()`?  Both these methods require configuring the report before they can be invoked.  The solution is to write a `configure()` method that they will call (with `this.configure(options)`) that sets up the report for them.

Basically, the CashflowReportPage would now just be:
```js
class CashflowReportPage extends BaseReportPage {
  constructor() {
    super();
  }

  async configure(options) {
    await components.dateInterval.range(dateFrom, dateTo);
    await components.multipleCashBoxSelect.set(cashboxes);
  }
}
```
... and that is it.  The rest of the methods (`printPreview()`, `saveAsPDF()`, etc) will just use the `configure()` step to make sure the report is set up.
